exchange/utils.go

```diff
@@ -297,3 +309,105 @@ func randomizeList(list []openrtb_ext.BidderName) {
 		list[i], list[j] = list[j], list[i]
 	}
 }
+
+func applyRegs(bidRequest *openrtb.BidRequest, isAMP, applyGDPR, applyCOPPA bool) {
+	if bidRequest.User != nil {
+		// Need to duplicate pointer objects
+		user := *bidRequest.User
+		bidRequest.User = &user
+
+		// There's no way for AMP to send a GDPR consent string yet so it's hard
+		// to know if the vendor is consented or not and therefore for AMP requests
+		// we keep the BuyerUID as is
+		if !isAMP {
+			bidRequest.User.BuyerUID = ""
+		}
+		if applyCOPPA {
+			bidRequest.User.ID = ""
+			bidRequest.User.Yob = 0
+			bidRequest.User.Gender = ""
+			bidRequest.User.BuyerUID = ""
```

Comment:
We should probably also clean out any user mappings we have defined for the user.